File: Clothes_Store_Angular_Part/src/app/model/commandline.ts
import { Product } from "./product";

export class CommandLine {
  id: number;
  command: number;
  product: Product;
  size: string;
  quantity: number;

  constructor(product: Product, size: string, quantity: number) {
    this.product = product;
    this.size = size;
    this.quantity = quantity;
  }
}

File: Clothes_Store_Spring_Boot_Part/com.store/src/main/java/com/store/model/CommandLine.java
package com.store.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Objects;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "command_line")
public class CommandLine {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "command_id")
    private Command command;

    @ManyToOne
    @JoinColumn(name = "product_id")
    private Product product;

    @Column(name = "size")
    private String size;

    @Column(name = "quantity")
    private Long quantity;

    @Override
    public int hashCode() {
        return Objects.hash(id, size, quantity);
    }
}
